sql/pipelines/src/test/scala/org/apache/spark/sql/pipelines/graph/SqlPipelineSuite.scala

```diff
@@ -0,0 +1,788 @@
+                  |CREATE MATERIALIZED VIEW d AS
+                  |WITH c AS (
+                  | WITH b AS (
+                  |   SELECT * FROM a
+                  | )
+                  | SELECT * FROM b
+                  |)
+                  |SELECT * FROM c;
+                  |""".stripMargin
+    )
+
+    startPipelineAndWaitForCompletion(unresolvedDataflowGraph)
+
+    assert(
+      spark
+        .sql(s"SELECT * FROM ${fullyQualifiedIdentifier("d").quotedString}")
+        .collect()
+        .toSet == Set(1L).map(Row(_))
+    )
+  }
+
+  test("Unsupported SQL statements throws error") {
+    val graphRegistrationContext = new TestGraphRegistrationContext(spark)
+    val sqlGraphRegistrationContext = new SqlGraphRegistrationContext(graphRegistrationContext)
+
+    val ex = intercept[SqlGraphElementRegistrationException] {
+      sqlGraphRegistrationContext.processSqlFile(
+        sqlText = "CREATE TABLE t AS SELECT 1",
+        sqlFilePath = "a.sql",
+        spark = spark
+      )
+    }
+
+    assert(ex.getMessage.contains("Unsupported plan"))
+  }
+
+  test("Table schema is correctly parsed") {
+    val unresolvedDataflowGraph = unresolvedDataflowGraphFromSql(
+      sqlText = """
+                  |CREATE MATERIALIZED VIEW a (id LONG COMMENT 'comment') AS SELECT * FROM RANGE(5);
+                  |""".stripMargin
+    )
+
+    val resolvedDataflowGraph = unresolvedDataflowGraph.resolve()
+
+    // Let inferred/declared schema mismatch detection execute
+    resolvedDataflowGraph.validate()
+
+    val expectedSchema = new StructType().add(name = "id", dataType = LongType, nullable = false)
+
+    assert(
+      resolvedDataflowGraph.resolvedFlows
+        .find(_.identifier == fullyQualifiedIdentifier("a"))
+        .head
+        .schema == expectedSchema
+    )
+  }
+
+  test("Multipart table names supported") {
+    val database_name = "db_4159cf91_42c1_44d6_aa8c_9cd8a158230d"
+    val database2_name = "db_a90d194f_9dfd_44bf_b473_26727e76be7a"
+    spark.sql(s"CREATE DATABASE $database_name")
+    spark.sql(s"CREATE DATABASE $database2_name")
+
+    val unresolvedDataflowGraph = unresolvedDataflowGraphFromSql(
+      sqlText = s"""
+                   |CREATE MATERIALIZED VIEW $database_name.mv1 AS SELECT 1;
+                   |CREATE MATERIALIZED VIEW $database2_name.mv2 AS SELECT * FROM $database_name.mv1
+                   |""".stripMargin
+    )
+    startPipelineAndWaitForCompletion(unresolvedDataflowGraph)
+
+    assert(spark.sql(s"SELECT * FROM $database_name.mv1").collect().toSet == Set(1L).map(Row(_)))
+    assert(spark.sql(s"SELECT * FROM $database2_name.mv2").collect().toSet == Set(1L).map(Row(_)))
+  }
+
+  test("Flow cannot be created with multipart identifier") {
+    val graphRegistrationContext = new TestGraphRegistrationContext(spark)
+    val sqlGraphRegistrationContext = new SqlGraphRegistrationContext(graphRegistrationContext)
+
+    val ex = intercept[AnalysisException] {
```

Comment:
Besides analysis exception, I wonder what is the behavior on execution error. For example, a flow is to INSERT `1/col_a` and `col_a` contains `0`.

Reply:
I see. We can have a follow-up to add tests for execution failures.